### DISCUSSION TITLE
What is the purpose of the 'vals' returned from `tensor.find()`?

### DISCUSSION THREAD

**Discussion by DeepBlockDeepak** [sandialabs/pyttb#192] [General]
In the `tensor` class we see this behavior:
```python
>>> import pyttb as ttb
>>> import numpy as np
>>> X = ttb.tensor.from_data(3*np.random.rand(2,2,2)) # Generate some data.
>>> larger_entries = X > 2
>>> sv_tuple = larger_entries.find() # returns the subscripts of the nonzero values in X and a column vector of the values.
>>> sv_tuple
(array([[1, 1, 0],
        [1, 1, 1]]),
 array([[ True],
        [ True]]))
```
When `find()` is used directly on `X`, like in `X.find()`, the second element of the returned tuple does display the original tensor values. But in the context above, boolean values are produced instead. Is there utility in these boolean values over the actual values?

**Comment by ntjohnson1** [comment]
I don't know if I fully understand the question `larger_entries = X > 2` creates a boolean tensor based on the comparison. For that operation the only non-zero values are the true ones. However, for other tensors that might not be the case.
```python
>>> import pyttb as ttb
>>> X = ttb.tenzeros((4,4))
>>> X[0] = 99
>>> X[-1] = 55
>>> X.find()
(array([[0, 0],
        [3, 3]], dtype=int64),
 array([[99.],
        [55.]]))
```

**Reply by DeepBlockDeepak** [reply]
Oh I think I see now. I must have been mixing up `larger_entries` with a lambda which then gets used like a `filter` when `find()` is evoked, so I wasn't expecting to see boolean values in the result.
